src/DynamicDataVNext/Kernel/KeyedChangeSet.cs

```diff
@@ -56,6 +55,54 @@ public static class KeyedChangeSet
         };
     }
 
+    /// <summary>
+    /// Creates a new <see cref="KeyedChangeSet{TKey, TItem}"/> representing the addition of a range of items.
+    /// </summary>
+    /// <typeparam name="TKey">The type of the items' keys.</typeparam>
+    /// <typeparam name="TItem">The type of the items being added.</typeparam>
+    /// <param name="items">The items being added.</param>
+    /// <param name="keySelector">A selector for determining the key for each item being added.</param>
+    /// <returns>A <see cref="KeyedChangeSet{TKey, TItem}"/> describing the addition of the given items.</returns>
+    public static KeyedChangeSet<TKey, TItem> Addition<TKey, TItem>(
+        IEnumerable<TItem>  items,
```

Comment:
There are often use case, where the key cannot be selected by any consumer.  

Does the changeset API accommodate that?

Reply:
Sure, the types are freely-constructable, these are just convenience methods.